edx-solutions/xblock-drag-and-drop-v2
# [MCKIN-5776] Support for dynamic generation of target zones and image.

Instead of providing a custom image, the author can now specify the number of columns and rows and a single zone size and zones arranged in a grid of chosen size will be automatically generated, as well as the background image.

**Screenshots**:

![screen shot 2017-09-28 at 10 18 55](https://user-images.githubusercontent.com/32585/30958983-1d382c8e-a43f-11e7-9b11-c4a0680b91ab.png)


![screen shot 2017-09-28 at 10 19 17](https://user-images.githubusercontent.com/32585/30958985-1da99a2c-a43f-11e7-8299-cad8654252be.png)


![screen shot 2017-09-28 at 10 18 33](https://user-images.githubusercontent.com/32585/30958984-1d398610-a43f-11e7-849e-cf349ee60a89.png)


**Testing instructions**:

1. Verify that you are able to generate zones and the background image, that the auto generated background image saves successfully and that it works in the LMS.
1. Verify that you can switch between auto generated and manually provided image in the Studio without issues.
1. Verify in all supported browsers.

**Author notes and concerns**:

1. I was trying to test this out in IE10 (using crossbrowsertesting) where background image sizing didn't quite work correctly. I tried using a version of DnDv2 from before we started working on mobile tweaks, and the problem was still there. In IE11 the problem wouldn't even load for me in the Studio - again this does not seem to be a new issue.

**Reviewers**:

- [ ] @bradmerlin 
- [ ] @sanfordstudent
